client/ayon_core/pipeline/create/creator_plugins.py

```diff
@@ -22,6 +23,11 @@
 from .legacy_create import LegacyCreator
 
 
+if TYPE_CHECKING:
+    from .context import CreateContext, CreatedInstance, UpdateData
+    from ayon_core.lib import AbstractAttrDef
```

Comment:
> NameERROR: name 'AbstractAttrDef' is not defined

I faced the same error as mentioned in https://github.com/ynput/ayon-core/pull/366#pullrequestreview-2019802245
I don't know how to properly fix it but `AbstractAttrDef` wasn't imported anywhere and it's not imported because of the fact that `TYPE_CHECKING` is always `False` at run time therefore the interpreter can't find  `AbstractAttrDef`. 

I tweaked the import a little in order to test this PR as a user. _I can't tell if the type hints are correct or not because I don't know._

```suggestion
if TYPE_CHECKING:
    from .context import CreateContext, CreatedInstance, UpdateData
from ayon_core.lib import AbstractAttrDef
```